### PR TITLE
Move test utilities to utils/test folder

### PR DESCRIPTION
**Pull request by byrnHDF**
Cleanup utils checks

### CODE REVIEW

**Review by lrknox — APPROVED** [review]

File: tools/test/h5dump/CMakeVFDTests.cmake [hunk 1]
```diff
@@ -79,7 +66,7 @@ endmacro ()
 ##############################################################################
 
 # Run test with different Virtual File Driver
-foreach (vfd ${VFD_H5DUMP_LIST})
+foreach (vfd ${VFDP_LIST})
```
Comment (jhendersonHDF) [inline tools/test/h5dump/CMakeVFDTests.cmake]: VFDP -> VFD

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -378,6 +378,9 @@ set (H5_TESTS
     cmpd_dtransform
     event_set
 )
+if (HDF5_BUILD_UTILS)
+  set (H5_TESTS ${H5_TESTS} mirror_vfd)
+endif ()
```
Comment (jhendersonHDF) [inline test/CMakeLists.txt:383]: Why is this test only built based on the HDF5_BUILD_UTILS condition now? From a quick glance it didn't look like it was using anything from the utils dir.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -552,30 +555,25 @@ if (HDF5_ENABLE_FORMATTERS)
...
+    TARGET_C_PROPERTIES (mirror_vfd SHARED)
+    target_link_libraries (mirror_vfd PRIVATE ${HDF5_TEST_LIBSH_TARGET})
+  endif ()
+  set_target_properties (mirror_vfd PROPERTIES FOLDER test)
 
-#-----------------------------------------------------------------------------
-# Add Target to clang-format
-#-----------------------------------------------------------------------------
-if (HDF5_ENABLE_FORMATTERS)
-  clang_format (HDF5_TEST_event_set_FORMAT event_set)
```
Comment (jhendersonHDF) [inline test/CMakeLists.txt:578]: Looks like this line for the separate event_set test got accidentally removed

Comment (byrnHDF) [reply test/CMakeLists.txt:578]: Not accident it is included in the make macro as event_set is in the list

File: test/CMakeLists.txt (hunk 2 above)
Comment (byrnHDF) [reply test/CMakeLists.txt:383]: mirror_vfd needs the mirror_server in utils right?

Comment (jhendersonHDF) [reply test/CMakeLists.txt:383]: Ah ok, yes that's probably the case.

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**